src/common/ipaddr.cc

```diff
@@ -60,7 +60,7 @@ const struct ifaddrs *find_ipv4_in_subnet(const struct ifaddrs *addrs,
     if (addrs->ifa_addr == NULL)
       continue;
 
-    if (boost::starts_with(addrs->ifa_name, "lo:"))
+    if (strcmp(addrs->ifa_name, "lo") == 0 || boost::starts_with(addrs->ifa_name, "lo:"))
```

Comment:
@thomasgoirand @dvanders i think this practically reverts 201b59204374ebdab91bb554b986577a97b19c36. how about adding an "network_interfaces_excludes" setting? which defaults to "^lo$;^lo:"?

@jdurgin if we are in a hurry to get this change in the pacific backport. we can have it as a follow-up change.

Reply:
yes, let's handle that in a follow-up. we want to get the pacific and nautilus backports done to avoid releasing / fix the regression

Reply:
@tchaikov do we really need regexs? how about just a `exclude_loopback_interfaces` true/false which we can add the beginning of this `if`, defaulting to `true`?

Reply:
@dvanders no, that'd work also. i am working on a change implementing your suggestion.

Reply:
@tchaikov I also wonder if `cluster_network_interface` and `public_network_interface` could already be used if someone needs to bind to `lo`.

Reply:
I always thinking this better idea, because any application may be `binds to addr`(e.g. 127.0.0.1:8080) or `binds to interfaces` (e.g. vlan1144,vlan1249,lo and some port in other option obviously)

Reply:
@dvanders @jdurgin #40435